client/consensus/src/collators/lookahead.rs

```diff
@@ -636,7 +637,14 @@ where
                                 let slot = inherent_providers.slot();
                                 let container_chain_slot_duration = (params.get_current_slot_duration)(parent_header.hash());
 
-                                match try_to_buy_core::<_, _, <<OBlock as BlockT>::Header as HeaderT>::Number, _, CIDP, _, _>(params.para_id, aux_data, inherent_providers, &params.keystore, params.orchestrator_client.clone(), params.orchestrator_tx_pool.clone(), parent_header, params.orchestrator_slot_duration, container_chain_slot_duration).await {
+                                let buy_core_result = if params.solochain {
+                                    // TODO: implement parathread support for solochain
+                                    log::warn!("Unimplemented: cannot buy core for parathread in solochain");
```

Comment:
I would prefer if we can remove this panic. Since, as of now, it is possible to register parathreads in the solo chain using sudo, which can potentially trigger this. 

I would suggest instead of panic, if we can make the loop `no-op` by `continue` on next iteration?

Reply:
*duplicate*